// math.solver/index.js
let Add_Func = require('./Main/Methods/Basic/Add'),
    Substract_Func = require('./Main/Methods/Basic/Substract'),
    Multiply_Func = require('./Main/Methods/Basic/Multiply'),
    Divide_Func = require('./Main/Methods/Basic/Divide'),
      SquareRoot_Func = require('./Main/Methods/Advanced/SquareRoot'),
      CubicRoot_Func = require('./Main/Methods/Advanced/CubicRoot'),
      ExponentPower_Func = require('./Main/Methods/Advanced/ExponentPower')

class Calculator {
    constructor() {
    }

    add(number, plus) {
        return Add_Func(number, plus)
    }
    
    substract(number, substrate) {
        return Substract_Func(number, substrate)
    }

    multiply(number, multiplier) {
        return Multiply_Func(number, multiplier)
    }

    divide(number, divider) {
        return Divide_Func(number, divider)
    }
       
      square_root(number) {
          return SquareRoot_Func(number)
      }

      cubic_root(number) {
          return CubicRoot_Func(number)
      }

      power(number, power) {
          return ExponentPower_Func(number, power)
      }
}

module.exports = {
    Calculator
}
